src/libs/SelectionScraper/index.ts

```diff
@@ -96,58 +98,61 @@ const getHTMLOfSelection = () => {
 
 /**
  * Clears all attributes from dom elements
- * @param {Object} dom htmlparser2 dom representation
- * @param {Boolean} isChildOfEditorElement
- * @returns {Object} htmlparser2 dom representation
  */
-const replaceNodes = (dom, isChildOfEditorElement) => {
-    let domName = dom.name;
-    let domChildren;
-    const domAttribs = {};
-    let data;
+const replaceNodes = (dom: Node, isChildOfEditorElement: boolean): Node => {
+    const domElement = dom as Element;
+    let domName = domElement.name;
+    let domChildren: Node[] = [];
+    const domAttribs: Element['attribs'] = {};
+    let data = '';
 
     // Encoding HTML chars '< >' in the text, because any HTML will be removed in stripHTML method.
-    if (dom.type === 'text') {
-        data = Str.htmlEncode(dom.data);
+    if (dom.type.toString() === 'text') {
+        data = Str.htmlEncode((dom as DataNode).data);
     }
 
     // We are skipping elements which has html and body in data-testid, since ExpensiMark can't parse it. Also this data
     // has no meaning for us.
-    if (dom.attribs && dom.attribs[tagAttribute]) {
-        if (!elementsWillBeSkipped.includes(dom.attribs[tagAttribute])) {
-            domName = dom.attribs[tagAttribute];
+    if (domElement.attribs?.[tagAttribute]) {
+        if (!elementsWillBeSkipped.includes(domElement.attribs[tagAttribute])) {
+            domName = domElement.attribs[tagAttribute];
         }
-    } else if (dom.name === 'div' && dom.children.length === 1 && isChildOfEditorElement) {
+    } else if (domElement.name === 'div' && domElement.children.length === 1 && isChildOfEditorElement) {
         // We are excluding divs that are children of our editor element and have only one child to prevent
         // additional newlines from being added in the HTML to Markdown conversion process.
-        return replaceNodes(dom.children[0], isChildOfEditorElement);
+        return replaceNodes(domElement.children[0], isChildOfEditorElement);
     }
 
     // We need to preserve href attribute in order to copy links.
-    if (dom.attribs && dom.attribs.href) {
-        domAttribs.href = dom.attribs.href;
+    if (domElement.attribs?.href) {
+        domAttribs.href = domElement.attribs.href;
+    }
+
+    if (domElement.children) {
+        domChildren = domElement.children.map((c) => replaceNodes(c, isChildOfEditorElement || !!domElement.attribs?.[tagAttribute]));
     }
 
-    if (dom.children) {
-        domChildren = _.map(dom.children, (c) => replaceNodes(c, isChildOfEditorElement || !_.isEmpty(dom.attribs && dom.attribs[tagAttribute])));
+    if (data) {
+        return {
+            ...dom,
+            data,
+        } as DataNode;
     }
 
     return {
         ...dom,
-        data,
         name: domName,
         attribs: domAttribs,
         children: domChildren,
-    };
+    } as Element;
```

Comment:
I didn't fully understand why this change was made.

Reply:
I made this change because `replaceNodes` function should return `Node` type elements. `Node` type is a general type and there are many more specific ones that extend it. The problem was that children can change different types of Nodes. According to the `domhandler` types there isn't any type that have both `data` and `attribs`/`children` attributes. So if there is Node have `data` I'm returning `DataNode`, if no I'm returning `Element` type Node